crates/matrix-sdk-common/src/deserialized_responses.rs

```diff
@@ -703,6 +703,36 @@ impl From<SyncTimelineEventDeserializationHelperV0> for SyncTimelineEvent {
     }
 }
 
+/// Represent a failed to send unrecoverable error of an event sent via the
+/// send_queue. It is a serializable representation of a client error, see
+/// `From` implementation for more details. These errors can not be
+/// automatically retried, but yet some manual action can be taken before retry
+/// sending. If not the only solution is to delete the local event.
+#[derive(Clone, Debug, Serialize, Deserialize, thiserror::Error)]
+pub enum QueueWedgeError {
+    /// This error occurs when there are some insecure devices in the room, and
+    /// the current encryption setting prohibit sharing with them.
+    #[error("There are insecure devices in the room")]
```

Comment:
With this error message, you won't see the `user_device_map` in the display impl. Saying here in case it's not intended; to display them, you'd need to add something like `: {user_device_map:?}` to the string above, and it will include the debug display of the btreemap.